lib/IRGen/GenClass.cpp

```diff
@@ -503,13 +520,20 @@ ClassTypeInfo::createLayoutWithTailElems(IRGenModule &IGM,
 }
 
 const ClassLayout &
-ClassTypeInfo::getClassLayout(IRGenModule &IGM, SILType classType) const {
+ClassTypeInfo::getClassLayout(IRGenModule &IGM, SILType classType,
+                              bool completelyFragileLayout) const {
+  // Perform fragile layout only if Objective-C interop is enabled.
+  completelyFragileLayout &= IGM.Context.LangOpts.EnableObjCInterop;
+  completelyFragileLayout &= !IGM.IRGen.Opts.EnableClassResilience;
```

Comment:
This part seems weird. Shouldn't it be the caller's responsibility to ask for the correct layout? Maybe this can be an assertion instead, or maybe the parameter should be renamed "forStaticMetadata" or something.

Reply:
There are a bunch of callers, but most just pass 'false'. I think only two pass 'true'. Do you want them both to pass the right condition instead?

A change I was thinking of making here though was only ever doing the completely fragile layout if we stand to benefit from it, ie the class is not otherwise generic and doesn't have resilient _ancestry_, only resiliently-sized fields.

Reply:
I guess I'm inclined to change the name of the parameter, then.

Reply:
`preferFragileLayout`?

Reply:
The fragile layout is only used when emitting metadata, and even in a parallel universe without Objective-C, this could plausibly be part of some larger feature where we optimistically lay out all resilient data based on compile-time knowledge and then fix it up at runtime. However that doesn't help me narrow down a name though :)

Reply:
That's why I said "forStaticMetadata" initially, though I'm sure "static" is not the best word.